ui/tests/acceptance/task-fs-test.js

```diff
@@ -129,6 +130,131 @@ module('Acceptance | task fs', function(hooks) {
+          ModTime: moment()
+            .subtract(1, 'year')
+            .format(),
+        },
+        {
+          Name: 'mmm-small-mid-file',
+          IsDir: false,
+          Size: 1919,
+          ModTime: moment()
+            .subtract(6, 'month')
+            .format(),
+        },
+        {
+          Name: 'zzz-med-new-file',
+          IsDir: false,
+          Size: 191900,
+          ModTime: moment().format(),
+        },
+        {
+          Name: 'aaa-big-old-directory',
+          IsDir: true,
+          Size: 19190000,
+          ModTime: moment()
+            .subtract(1, 'year')
+            .format(),
+        },
+        {
+          Name: 'mmm-small-mid-directory',
+          IsDir: true,
+          Size: 1919,
+          ModTime: moment()
+            .subtract(6, 'month')
+            .format(),
+        },
+        {
+          Name: 'zzz-med-new-directory',
+          IsDir: true,
+          Size: 191900,
+          ModTime: moment().format(),
+        },
+      ];
+    });
+
+    await FS.visitPath({ id: allocation.id, name: task.name, path: '/' });
+
+    assert.deepEqual(FS.directoryEntryNames(), [
+      'aaa-big-old-directory',
+      'mmm-small-mid-directory',
+      'zzz-med-new-directory',
+      'aaa-big-old-file',
+      'mmm-small-mid-file',
+      'zzz-med-new-file',
+    ]);
+
+    await FS.sortBy('Name');
+
+    assert.deepEqual(FS.directoryEntryNames(), [
+      'zzz-med-new-file',
+      'mmm-small-mid-file',
+      'aaa-big-old-file',
+      'zzz-med-new-directory',
+      'mmm-small-mid-directory',
+      'aaa-big-old-directory',
+    ]);
+
+    await FS.sortBy('ModTime');
+
+    assert.deepEqual(FS.directoryEntryNames(), [
+      'zzz-med-new-file',
+      'mmm-small-mid-file',
+      'aaa-big-old-file',
+      'zzz-med-new-directory',
+      'mmm-small-mid-directory',
+      'aaa-big-old-directory',
+    ]);
+
+    await FS.sortBy('ModTime');
+
+    assert.deepEqual(FS.directoryEntryNames(), [
+      'aaa-big-old-directory',
+      'mmm-small-mid-directory',
+      'zzz-med-new-directory',
+      'aaa-big-old-file',
+      'mmm-small-mid-file',
+      'zzz-med-new-file',
+    ]);
+
+    await FS.sortBy('Size');
+
+    assert.deepEqual(
+      FS.directoryEntryNames(),
+      [
+        'aaa-big-old-file',
+        'zzz-med-new-file',
+        'mmm-small-mid-file',
+        'zzz-med-new-directory',
+        'mmm-small-mid-directory',
+        'aaa-big-old-directory',
+      ],
+      'expected files to be sorted by descending size and directories to be sorted by descending name'
+    );
+
+    await FS.sortBy('Size');
+
+    assert.deepEqual(
+      FS.directoryEntryNames(),
+      [
+        'aaa-big-old-directory',
+        'mmm-small-mid-directory',
+        'zzz-med-new-directory',
+        'mmm-small-mid-file',
+        'zzz-med-new-file',
+        'aaa-big-old-file',
+      ],
+      'expected directories to be sorted by name and files to be sorted by ascending size'
+    );
+  });
+
```

Comment:
I’ve also included a controller unit test version of this. I almost never write controller unit tests, but I thought it might be appropriate in this case rather than having a giant acceptance test. But it’s only a single test so it probably doesn’t slow it down too much… I’ll remove whichever one is less desirable!

Reply:
Yeah, this is an interesting one. The advantage of the acceptance test is it asserts not only that the sort works but that it is bound correctly to the template. 

If only we had routable components, then this could be an integration test 🙃

Reply:
hehe “controllers are going away” 😬